packages/grid/x-data-grid/src/hooks/features/selection/useGridSelection.ts

```diff
@@ -108,6 +112,9 @@ export const useGridSelection = (
         const visibleRowIds = gridVisibleSortedRowIdsSelector(apiRef);
         const startIndex = visibleRowIds.findIndex((rowId) => rowId === startId);
         const endIndex = visibleRowIds.findIndex((rowId) => rowId === endId);
```

Comment:
I don't think so because when expanding with the mouse, we support cross-page expansion.

You click on a row of page 1
You go to page 2
You shift click a row of page 2

=> It selects all the row between your two clicks.